cluster/addons.go

```diff
@@ -38,15 +44,113 @@ func (c *Cluster) deployK8sAddOns(ctx context.Context) error {
 func (c *Cluster) deployUserAddOns(ctx context.Context) error {
 	log.Infof(ctx, "[addons] Setting up user addons..")
 	if c.Addons == "" {
-		log.Infof(ctx, "[addons] No user addons configured..")
-		return nil
+		return c.deployAddonsInclude(ctx)
 	}
 
 	if err := c.doAddonDeploy(ctx, c.Addons, UserAddonResourceName); err != nil {
 		return err
 	}
 	log.Infof(ctx, "[addons] User addon deployed successfully..")
+	return c.deployAddonsInclude(ctx)
+}
+
+// deployAddonsInclude takes any passed addons from the addons key and appends yaml contents from urls/paths
+// then deploys all addons at once. Not doing this in this manner will fail to deploy all of the plugins.
+// Only the addons from the addons: key would be deployed.
+func (c *Cluster) deployAddonsInclude(ctx context.Context) error {
+	var userAddons = ""
+	log.Infof(ctx, "[addons] Checking for included user addons")
+
+	yamlDelimiter := regexp.MustCompile("(\n---$|\n---\n$)")
+	// If there are contents in the array continue
+	if len(c.AddonsInclude) != 0 {
```

Comment:
can we revers this ? as in:
```
if len(c.AddonsInclude) == 0 {
    log.Infof(ctx, "[addons] No included addon paths or urls..")
    return nil
}
```
Makes the code a bit simpler.

Reply:
Sure thing